game-core/src/main/java/games/strategy/engine/framework/GameDataManager.java

```diff
@@ -93,39 +94,28 @@ public static GameData loadGame(final InputStream is) throws IOException {
                 version,
                 UrlConstants.DOWNLOAD_WEBSITE,
                 UrlConstants.DOWNLOAD_WEBSITE));
+        return Optional.empty();
       } else if (!HeadlessGameServer.headless()
           && ((Version) version).isGreaterThan(ClientContext.engineVersion())) {
-        // we can still load it because our engine is compatible, however this save was made by a
-        // newer engine, so prompt the user to upgrade
-        promptToLoadNewerSaveGame();
+        // Prompt the user to upgrade
+        log.warn(
+            "This save was made by a newer version of TripleA.<br>"
+                + "To load this save, download the latest version of TripleA: "
+                + "<a href=\"{}\">{}</a>",
+            UrlConstants.DOWNLOAD_WEBSITE,
+            UrlConstants.DOWNLOAD_WEBSITE);
+        return Optional.empty();
+      } else {
+        final GameData data = (GameData) input.readObject();
+        data.postDeSerialize();
+        loadDelegates(input, data);
+        return Optional.of(data);
       }
-
-      final GameData data = (GameData) input.readObject();
-      data.postDeSerialize();
-      loadDelegates(input, data);
-      return data;
     } catch (final ClassNotFoundException cnfe) {
-      throw new IOException(cnfe.getMessage());
-    }
-  }
-
-  private static void promptToLoadNewerSaveGame() throws IOException {
-    // this is needed because a newer client might depend on variables that are part of the new
-    // save but will be stripped by the old client. When the old client re-saves the data, the
-    // new client will load the save game and be in odd state.
-    final String messageString =
-        "This save was made by a newer version of TripleA."
-            + "\nPlaying newer saves with an older engine can lead to unpredictable problems. "
-            + "It is recommended that you upgrade to the latest version of TripleA before playing "
-            + "this save game. To download the latest version of TripleA, Please visit "
-            + UrlConstants.DOWNLOAD_WEBSITE
-            + ".\n"
-            + "\n\nDo you wish to continue and open this save with your current 'old' version?;";
-    final int answer =
-        JOptionPane.showConfirmDialog(
-            null, messageString, "Open Newer Save Game?", JOptionPane.YES_NO_OPTION);
-    if (answer != JOptionPane.YES_OPTION) {
-      throw new IOException("Loading the save game was aborted");
+      log.error("Error loading game data", cnfe);
+      return Optional.empty();
+    } catch (final IOException e) {
+      return Optional.empty();
```

Comment:
Should a `log.error` be added here as well?

Reply:
Yes, an error of omission on my part. That was something I had intended to come back to, updated in: c7ae2cac3